examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp

```diff
@@ -25,67 +25,93 @@
 #include <fastrtps/utils/eClock.h>
 #include <fastrtps/log/Log.h>
 
+#include <string>
+
 using namespace eprosima;
 using namespace fastrtps;
 using namespace rtps;
 int main(int argc, char** argv)
 {
-    std::cout << "Starting "<< std::endl;
+    std::cout << "Starting " << std::endl;
     int type = 1;
     int count = 0;
     long sleep = 100;
+    std::string wan_ip;
+    int port = 5100;
 
-    if(argc > 1)
+    if (argc > 1)
     {
-        if(strcmp(argv[1],"publisher")==0)
+        if (strcmp(argv[1], "publisher") == 0)
         {
             type = 1;
             if (argc >= 3)
             {
                 count = atoi(argv[2]);
-                if (argc == 4)
+                if (argc > 3)
                 {
                     sleep = atoi(argv[3]);
                 }
+
+                if (argc > 4)
+                {
+                    wan_ip = std::string(argv[4]);
+                }
+
+                if (argc > 5)
+                {
+                    port = atoi(argv[5]);
+                }
             }
         }
-        else if(strcmp(argv[1],"subscriber")==0)
+        else if (strcmp(argv[1], "subscriber") == 0)
         {
             type = 2;
+
+            if (argc > 2)
+            {
+                wan_ip = std::string(argv[2]);
+            }
+
+            if (argc > 3)
+            {
+                port = atoi(argv[3]);
+            }
         }
     }
     else
     {
         std::cout << "publisher OR subscriber argument needed" << std::endl;
-        std::cout << "publisher has optional arguments: publisher [times] [interval] " << std::endl;
+        std::cout << "publisher has optional arguments: publisher [times] [interval] [wan_ip] [port] " << std::endl;
```

Comment:
Add info line about subscriber optional arguments.

Reply:
Updated with a small explanation of the arguments